test/e2e/apimachinery/webhook.go

```diff
@@ -782,23 +786,76 @@ func registerWebhookForWebhookConfigurations(f *framework.Framework, context *ce
 	}
 }
 
+func registerMutatingWebhookForWebhookConfigurations(f *framework.Framework, context *certContext) func() {
+	var err error
+	client := f.ClientSet
+	By("Registering a mutating webhook on ValidatingWebhookConfiguration and MutatingWebhookConfiguration objects, via the AdmissionRegistration API")
+
+	namespace := f.Namespace.Name
+	configName := mutatingWebhookForWebhooksConfigName
+	failurePolicy := v1beta1.Fail
+
+	// This webhook adds a label to all requests create to validating webhook configuration and
+	// mutating webhook configuration objects. It should never be called, however, because
+	// dynamic admission webhooks should not be called on requests involving webhook configuration objects.
+	_, err = client.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Create(&v1beta1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: configName,
+		},
+		Webhooks: []v1beta1.Webhook{
+			{
+				Name: "add-label-to-webhook-configurations.k8s.io",
+				Rules: []v1beta1.RuleWithOperations{{
+					Operations: []v1beta1.OperationType{v1beta1.Create},
+					Rule: v1beta1.Rule{
+						APIGroups:   []string{"admissionregistration.k8s.io"},
+						APIVersions: []string{"*"},
+						Resources: []string{
+							"validatingwebhookconfigurations",
+							"mutatingwebhookconfigurations",
+						},
+					},
+				}},
+				ClientConfig: v1beta1.WebhookClientConfig{
+					Service: &v1beta1.ServiceReference{
+						Namespace: namespace,
+						Name:      serviceName,
+						Path:      strPtr("/add-label"),
+					},
+					CABundle: context.signingCert,
+				},
+				FailurePolicy: &failurePolicy,
+			},
+		},
+	})
+	framework.ExpectNoError(err, "registering webhook config %s with namespace %s", configName, namespace)
+
+	// The webhook configuration is honored in 10s.
+	time.Sleep(10 * time.Second)
+	return func() {
```

Comment:
Might be nice to write a proper DeleteAndWaitForCleanup method. It can be built on the waitPool() method which should shorten the test run time.

Reply:
Sure, that would be better. I think this was just copied from the other tests.